pilot/pkg/model/policyattachment.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"istio.io/api/type/v1beta1"
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/config/schema/gvk"
+)
+
+type policyTargetGetter interface {
+	GetTargetRef() *v1beta1.PolicyTargetReference
+	GetSelector() *v1beta1.WorkloadSelector
+}
+
+type policyMatch string
+
+const (
+	policyMatchSelector policyMatch = "selector"
+	policyMatchDirect   policyMatch = "direct"
+	policyMatchIgnore   policyMatch = "ignore"
+)
+
+func getPolicyMatcher(policyName string, opts workloadSelectionOpts, policy policyTargetGetter) policyMatch {
+	gatewayName, isGatewayAPI := opts.workloadLabels[constants.IstioGatewayLabel]
+	targetRef := policy.GetTargetRef()
+	if isGatewayAPI && targetRef == nil && policy.GetSelector() != nil {
+		if opts.isWaypoint || features.EnableGatewayPolicyAttachmentOnly {
```

Comment:
The flag is not obvious, it means EnableOnlyPolicyAttachment on gateway api

Reply:
Do you have ideas for clearer names?

Reply:
How about EnableSelectorBasedK8sGatewayPolicy,

Reply:
Keep in mind the defaults; because bools have a zero value of false, I picked a name that indicates the opposite of the new behavior. To clarify, if the flag is set to true, then workloadSelector policies that select the gateway label WILL NOT APPLY. Put another way, the flag forces users to apply a targetRef

Reply:
The intention is to indicate whether a policy with a selector can apply to a k8s gateway. 
I do not think bool type matters here.

TBH I think `EnableGatewayPolicyAttachmentOnly ` is more clearer than the negative naming `EnableGatewayPolicyAttachmentOnly `.

Reply:
Did you mean EnableSelectorBasedK8sGatewayPolicy for the first one? The more I think about it, I think I agree with you; will rename